servers/cu/src/domain/client/ao-process.test.js

```diff
@@ -1396,5 +1396,107 @@ describe('ao-process', () => {
+      const saveCheckpoint = saveCheckpointWith({
+        ...depsAll,
+        address: () => {
+          return Promise.resolve()
+        },
+        queryGateway: () => Promise.resolve({ data: { transactions: { edges: [{ node: { id: 'tx-123', tags: [] } }] } } }),
+        queryCheckpointGateway: () => Promise.resolve({ data: { transactions: { edges: [{ node: { id: 'tx-123', tags: [] } }] } } }),
+        readProcessMemoryFile: () => {
+          assert.fail('should not attempt to read the process memory file')
+        },
+        writeCheckpointRecord: () => Promise.resolve(),
+        DISABLE_PROCESS_CHECKPOINT_CREATION: false
+      })
+      const result = await saveCheckpoint({ File: Promise.resolve('file') })
+      assert.equal(result.id, 'tx-123', 'should return the id of the checkpoint')
+    })
+
+    describe('should fail if the memory is not provided directly or as a field path', () => {
+      const deps = {
+        ...depsAll,
+        address: () => {
+          return Promise.resolve()
+        },
+        queryGateway: () => Promise.resolve(),
+        queryCheckpointGateway: () => Promise.resolve(),
+        DISABLE_PROCESS_CHECKPOINT_CREATION: false
+      }
+      test('should fail if memory and file are both undefined', async () => {
+        const saveCheckpoint = saveCheckpointWith({
+          ...deps
+        })
+        try {
+          await saveCheckpoint({})
+        } catch (e) {
+          assert.equal(e, 'either File or Memory required')
+        }
+      })
+      test('should fail if file promise resolves with undefined', async () => {
+        const saveCheckpoint = saveCheckpointWith({
+          ...deps
+        })
+        try {
+          await saveCheckpoint({ File: Promise.resolve(undefined) })
+        } catch (e) {
+          assert.equal(e, 'either Memory or File required')
+        }
+      })
+    })
+
+    test('should save a checkpoint', async () => {
+      const saveCheckpoint = saveCheckpointWith({
+        ...depsAll,
+        queryGateway: () => Promise.resolve(),
+        queryCheckpointGateway: () => Promise.resolve(),
+        address: () => Promise.resolve(),
+        hashWasmMemory: () => Promise.resolve('hash'),
+        readProcessMemoryFile: () => Promise.resolve(Buffer.from([1, 2, 3])),
+        uploadDataItem: (data) => Promise.resolve({ id: 'tx-123' }),
+        buildAndSignDataItem: (...args) => Promise.resolve(...args),
+        writeCheckpointRecord: (d) => {
+          assert.ok(true)
+          return Promise.resolve()
+        },
+        DISABLE_PROCESS_CHECKPOINT_CREATION: false
+      })
+      try {
+        const result = await saveCheckpoint({ File: 'file', moduleId: 'module-123', processId: 'process-123' })
+        assert.equal(result.id, 'tx-123', 'should return the id of the checkpoint')
+      } catch (e) {
+        console.log('error', e)
+      }
```

Comment:
Catching an error here will cause the test to always pass, even if an error thrown. We should remove the `try/catch`